core/trino-main/src/test/java/io/trino/cost/TestFilterStatsCalculator.java

```diff
@@ -205,6 +207,123 @@ public void testComparison()
         }
     }
 
+    @Test
+    public void testInequalityComparisons()
+    {
+        double inputRowCount = standardInputStatistics.getOutputRowCount();
```

Comment:
please add some test cases with `NaN` and `Infinites` (min, max, null fraction, Ndv)

Reply:
I've added tests to TestComparisonStatsCalculator